src/fieldAnalyser/utils.ts

```diff
@@ -0,0 +1,34 @@
+import PlatformClient, {FieldModel} from '@coveord/platform-client';
+
+export const listAllFieldsFromOrg = async (
+  client: PlatformClient,
+  page = 0,
+  fields: FieldModel[] = []
+): Promise<string[]> => {
+  const list = await client.field.list({
+    page,
+    perPage: 1000,
+  });
+
+  fields.push(...list.items);
+
+  if (page < list.totalPages - 1) {
+    return listAllFieldsFromOrg(client, page + 1, fields);
+  }
+
+  return fields.map(({name}) => `${name}`);
+};
+
+export const createFields = async (
+  client: PlatformClient,
+  fields: FieldModel[],
+  fieldBatch = 500
+) => {
+  if (fields.length === 0) {
+    return;
+  }
```

Comment:
This is superfluous I think: the first evaluation of the for condition will fail, which mean it'll be skipped

Reply:
No it's important otherwise we get an error from the API when trying to upload an empty list of fields